typing/includecore.ml

```diff
@@ -102,9 +106,17 @@ let value_descriptions ~loc env name
   | Ok () -> ()
   | Error e -> raise (Dont_match (Zero_alloc e))
   end;
-  begin match Mode.Modality.Value.sub vd1.val_modalities vd2.val_modalities with
-  | Ok () -> ()
-  | Error e -> raise (Dont_match (Modality e))
+  (* [wrap_constraint_with_shape] invokes inclusion check with idential inferred
+    modalities, which we need to workaround. *)
+  if not (vd1.val_modalities == vd2.val_modalities) then begin
```

Comment:
Is this workaround new or was it previously in `Modality.Value.sub`?

Reply:
Yes it's also in `Modality.Value.sub`, which is probably unreachable now.